examples/model_selection/plot_grid_search_text_feature_extraction.py

```diff
@@ -0,0 +1,235 @@
+"""
+==========================================================
+Sample pipeline for text feature extraction and evaluation
+==========================================================
+
+The dataset used in this example is :ref:`20newsgroups_dataset` which will be
+automatically downloaded, cached and reused for the document classification
+example.
+
+In this example we tune the hyperparameters of a particular classifier using a
+:class:`~sklearn.model_selection.RandomizedSearchCV`. For a demo on the
+performance of some other classifiers, see the
+:ref:`sphx_glr_auto_examples_text_plot_document_classification_20newsgroups.py`
+notebook.
+
+"""
+
+# Author: Olivier Grisel <olivier.grisel@ensta.org>
+#         Peter Prettenhofer <peter.prettenhofer@gmail.com>
+#         Mathieu Blondel <mathieu@mblondel.org>
+#         Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+# License: BSD 3 clause
+
+# %%
+# Data loading
+# ------------
+# We load two categories from the training set. You can adjust the number of
+# categories by adding their names to the list or setting `categories=None` in
+# the dataset loader to get the 20 of them.
+
+from sklearn.datasets import fetch_20newsgroups
+
+categories = [
+    "alt.atheism",
+    "talk.religion.misc",
+]
+
+data_train = fetch_20newsgroups(
+    subset="train",
+    categories=categories,
+    shuffle=True,
+    random_state=42,
+    remove=("headers", "footers", "quotes"),
+)
+
+data_test = fetch_20newsgroups(
+    subset="test",
+    categories=categories,
+    shuffle=True,
+    random_state=42,
+    remove=("headers", "footers", "quotes"),
+)
+
+print(f"Loading 20 newsgroups dataset for {len(data_train.target_names)} categories:")
+print(data_train.target_names)
+print(f"{len(data_train.data)} documents")
+
+# %%
+# Pipeline with hyperparameter tuning
+# -----------------------------------
+# We define a pipeline combining a text feature vectorizer with a simple
+# classifier.
```

Comment:
```suggestion
# classifier yet effective for text classification.
```

Reply:
Do you mean efficient?
```suggestion
# yet efficient classifier for text classification.
```

Reply:
effective seems to do the job :)

<img width="657" alt="image" src="https://user-images.githubusercontent.com/7454015/189855679-4ce78a87-b37d-46b8-8726-5201601812c5.png">